src/components/Homepage.jsx

```diff
@@ -1,33 +1,83 @@
 import React, { useState, useEffect } from 'react';
 import { useSearchParams } from 'react-router-dom';
-import ArticleList from "./ArticleList";
-import { getArticles } from "../api";
-
-export function Homepage( {filterTopic}) {
-    const [articles, setArticles] = useState([]);
-    const [loading, setLoading] = useState(true);
-    const [searchParams] = useSearchParams();
-    const topic = searchParams.get('topic') || filterTopic
-
-    useEffect(() => {
-        setLoading(true);
-        getArticles(topic)
-          .then(data => {
-            setArticles(data.articles);
-            setLoading(false);
-          })
-          .catch(error => {
-            console.error('Error fetching articles:', error);
+import ArticleList from './ArticleList';
+import { getArticles } from '../api';
+
+export function Homepage() {
+  const [articles, setArticles] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const topic = searchParams.get('topic') || '';
+  const sortByParam = searchParams.get('sortBy') || 'created_at';
+  const sortOrderParam = searchParams.get('order') || 'desc';
+
+  const [sortBy, setSortBy] = useState(sortByParam);
+  const [sortOrder, setSortOrder] = useState(sortOrderParam);
+
+  useEffect(() => {
+    setLoading(true);
+    getArticles(topic)
+      .then(data => {
+        let sortedArticles = data.articles;
+        if (sortBy) {
+          sortedArticles = sortedArticles.sort((a, b) => {
+            let valueA = a[sortBy];
+            let valueB = b[sortBy];
+
+            // Convert comment_count to number for accurate sorting
+            if (sortBy === 'comment_count') {
+              valueA = Number(a[sortBy]);
+              valueB = Number(b[sortBy]);
+            }
+
+            if (sortOrder === 'asc') {
+              return valueA > valueB ? 1 : -1;
+            } else {
+              return valueA < valueB ? 1 : -1;
+            }
```

Comment:
Ideally you'd get your back end to sort this for you instead of handling it on the front end, but this depends on whether your back end is built to be able to do this